packages/astro/src/core/preview/index.ts

```diff
@@ -113,15 +113,18 @@ export default async function preview(
 			const listen = () => {
 				httpServer = server.listen(port, host, async () => {
 					if (!showedListenMsg) {
-						const devServerAddressInfo = server.address() as AddressInfo;
+						const resolvedUrls = msg.resolveServerUrls({
```

Comment:
Ah, it' a shame or crazy `ipv4` code needs to stick around for preview servers. I'm curious if Vite could expose its internal `resolveUrls` helper for this, similar to `normalizePath` and `mergeConfig`?

Reply:
Yeah this sounds like a good idea! Since the Vite preview server is more limited, it make sense to expose this part out.